src/main/java/org/mitre/dsmiley/httpproxy/ProxyServlet.java

```diff
@@ -258,17 +257,39 @@ protected void initTarget() throws ServletException {
    * In any case, it should be thread-safe.
    */
   protected HttpClient createHttpClient() {
-    HttpClientBuilder clientBuilder = HttpClientBuilder.create()
+    HttpClientBuilder clientBuilder = getHttpClientBuilder()
                                         .setDefaultRequestConfig(buildRequestConfig())
                                         .setDefaultSocketConfig(buildSocketConfig());
     
     clientBuilder.setMaxConnTotal(maxConnections);
     
     if (useSystemProperties)
       clientBuilder = clientBuilder.useSystemProperties();
+    return buildHttpClient(clientBuilder);
+  }
+
+  /**
+   * Creates a HttpClient from the given builder. Meant as postprocessor
+   * to possibly adapt the client builder prior to creating the
+   * HttpClient.
+   *
+   * @param clientBuilder pre-configured client builder
+   * @return HttpClient
+   */
+  protected HttpClient buildHttpClient(HttpClientBuilder clientBuilder) {
```

Comment:
I wonder how useful this is... instead couldn't a user subclass and override createHttpClient and then call super.createHttpClient if they choose?

Reply:
There is a slight difference regarding the two behaviors: While the `super` approach would only allow to work on a ready built `HttpClient` you can influence the `HttpClientBuilder` with the suggested approach.

So, having `getHttpClientBuilder` and `buildHttpClient` allows pre- as well as post-processing. This is actually what to argue about. For the adaptions we required, modifying `getHttpClientBuilder` would have been enough (to set the SSLSocketFactory). Doing such an override in `buildHttpClient` just provides more confidence that the default behavior does not override this factory again (for example).